distribution/tools/server-cli/src/main/java/org/elasticsearch/server/cli/ServerCli.java

```diff
@@ -80,25 +79,31 @@ public void execute(Terminal terminal, OptionSet options, Environment env, Proce
 
         validateConfig(options, env);
 
-        // setup security
-        final SecureString keystorePassword = getKeystorePassword(env.configFile(), terminal);
-        env = autoConfigureSecurity(terminal, options, processInfo, env, keystorePassword);
+        try (KeyStoreWrapper keystore = KeyStoreWrapper.load(env.configFile())) {
+            // setup security
+            final SecureString keystorePassword = getKeystorePassword(keystore, terminal);
+            env = autoConfigureSecurity(terminal, options, processInfo, env, keystorePassword);
 
-        // install/remove plugins from elasticsearch-plugins.yml
-        syncPlugins(terminal, env, processInfo);
+            if (keystore != null) {
+                keystore.decrypt(keystorePassword.getChars());
+            }
 
-        ServerArgs args = createArgs(options, env, keystorePassword, processInfo);
-        this.server = startServer(terminal, processInfo, args, env.pluginsFile());
+            // install/remove plugins from elasticsearch-plugins.yml
+            syncPlugins(terminal, env, processInfo);
 
-        if (options.has(daemonizeOption)) {
-            server.detach();
-            return;
-        }
+            ServerArgs args = createArgs(options, env, keystore, keystorePassword, processInfo);
+            this.server = startServer(terminal, processInfo, args);
 
-        // we are running in the foreground, so wait for the server to exit
-        int exitCode = server.waitFor();
-        if (exitCode != ExitCodes.OK) {
-            throw new UserException(exitCode, "Elasticsearch exited unexpectedly");
+            if (options.has(daemonizeOption)) {
```

Comment:
The keystore should be closed once we reach here. The detach/wait should happen outside of the try.